kernel/thread.c

```diff
@@ -270,6 +270,16 @@ void _setup_new_thread(struct k_thread *new_thread,
 		       void *p1, void *p2, void *p3,
 		       int prio, u32_t options)
 {
+#if CONFIG_STACK_POINTER_RANDOM
+#if defined(STACK_GROWS_UP)
+	/* This is so rare not bothering for now */
+#error "Stack pointer randomization not implemented for upward growing stacks"
+#endif
+	/* Don't need to worry about alignment of the size here, _new_thread()
+	 * is required to do it
+	 */
+	stack_size -= sys_rand32_get() % CONFIG_STACK_POINTER_RANDOM;
```

Comment:
I worry that using the MOD operator here will bias the offset towards a predictable value... specially since sys_rand32_get() does not offer any distribution guarantees, the range of possible random values depends on the sys_rand32_get() implementation, and CONFIG_STACK_POINTER_RANDOM is a user-defined value.

A good way to fix this is to separate the random number generator space into `CONFIG_STACK_POINTER_RANDOM` bins and keep generating numbers if they don't fit in any of those bins.  Went looking for an example code in  a random library and stumbled upon this [Stackoverflow answer that seems pretty decent](https://stackoverflow.com/a/6852396).

This should be fixed at the random subsys level, of course, as there are other places in the kernel where we would at least have some better even distribution guarantees (e.g. stack canary generation) than what sys_rand32_get() offers.

In the meantime, please add a FIXME comment here.